addons/web/static/src/core/utils/html.js

```diff
@@ -286,3 +286,27 @@ export function setElementContent(element, content) {
         element.textContent = content;
     }
 }
+
+export function isMarkup(value) {
+    return value instanceof Markup;
+}
+
+/**
+ * Compares two objects or their string representations for equality.
+ * If one is a markup object and the other is not, they are considered unequal.
+ * @param {any} a
+ * @param {any} b
+ * @returns {boolean}
+ */
+export function isHtmlTextuallyEqual(a, b) {
+    if (isMarkup(a) && isMarkup(b)) {
+        return (
+            createElementWithContent("div", a).textContent ===
+            createElementWithContent("div", b).textContent
+        );
+    }
+    if (isMarkup(a) || isMarkup(b)) {
+        return false;
+    }
+    return String(a) === String(b);
+}
```

Comment:
I'm actually confused by this spec now.
It feels like with `textContent` specifically, we don't need a make a special case of string vs markup, as the end result would be the same (for simple string without tags).

Reply:
I dont feel like this is needed for now